R/data-storage-log-file.R

```diff
@@ -0,0 +1,182 @@
+#' Data storage class for JSON Log File
+#'
+#' @description
+#' Implementation of the DataStorage R6 class to a JSON log file backend using a unified
+#' API for read/write operations
+#'
+#' @export
+#'
+#' @examples
+#' data_storage <- DataStorageLogFile$new(
+#'   username = "test_user",
+#'   log_file_path = tempfile(pattern = "user_stats", fileext = ".json"),
+#'   session_file_path = tempfile(pattern = "session_details", fileext = ".json")
+#' )
+#' data_storage$insert(list(id = "an_id", action = "click"))
+#' data_storage$insert(list(id = "another_id", action = "click"))
+#' data_storage$read_user_data(as.Date("2020-01-01"), as.Date("2025-01-01"))
+DataStorageLogFile <- R6::R6Class( # nolint object_name_linter
+  classname = "DataStorageLogFile",
+  inherit = DataStorage,
+  #
+  # Public
+  public = list(
+    # @title initialize RSQLite storage provider
+    # @param username string with username of the current session
+    #
+
+    #' @description
+    #' Initialize the data storage class
+    #' @param username string with username of the current session
+    #' @param session_id string with custom session id (should not be used)
+    #' @param log_file_path string with path to JSON log file user actions
+    #' @param session_file_path string with path to JSON log file for the session details
+
+    initialize = function(
+    username, session_id = NULL, log_file_path, session_file_path
+    ) {
+      super$initialize(username, session_id)
+      logger::log_info("path to file: {log_file_path}")
+      private$connect(log_file_path = log_file_path, session_file_path = session_file_path)
+    },
```

Comment:
I opted for a two file approach here, one for user logs and another for session details. I also thought about using only one directory, but I think using two files make it more flexible.